src/components/batch-selection/__internal__/batch-selection.context.ts

```diff
@@ -0,0 +1,11 @@
+import React from "react";
+
+export type BatchSelectionContextType = {
+  disabled?: boolean;
```

Comment:
comment: we don't need to make this prop optional as it has a default value below

Reply:
suggestion (non-blocking): We could also rename the prop so that we don't have to rename it when we destructure.

```
const { disabled: batchSelectionDisabled } = useContext(
      BatchSelectionContext
    );
```

could be

```
const { batchSelectionDisabled } = useContext(
      BatchSelectionContext
    );
```